### ISSUE TITLE
Reconsider deprecation of `commitMessage`

### ISSUE THREAD

**Issue by shockey** [renovatebot/renovate#4128]
### This is a:

- [ ] Bug report (non-security related)
- [ ] Feature request
- [x] I'm not sure which of those it is

### I'm using:

- [x] The Renovate GitHub App
- [ ] Self-hosted GitHub
- [ ] Self-hosted GitLab
- [ ] Self-hosted Azure DevOps

### Please describe the issue:

We wish for our PRs/commit messages for npm dependency updates to follow a specific, compact pattern:
* minor/patch changes: `housekeeping: package-name@1.2.3`
* major changes: `housekeeping: package-name v2`

Our configuration is set up accordingly:

https://github.com/swagger-api/swagger-ui/blob/f73260bc5fb428a92c2462b351af9b9869732fbf/renovate.json#L5-L10

The tricky bit is that we can't model this without the deprecated `commitMessage` option, since Renovate's default for the option would always place a space in the middle of our minor/patch messages, i.e. `housekeeping: package-name @1.2.3`.

Is there a way to achieve this by only using the subfield options? I imagine it _may_ be possible to just stuff everything into either Topic or Extra, but I couldn't find any docs on scoping for the variables available to each field.

**Comment by HonkingGoose** [comment]
I noticed this issue is still around, might be good to make a decision now?

The latest Renovate config from their repository contains:

<details><summary>Click me to see their current Renovate config</summary>

```json
{
  "extends": [
    "config:base"
  ],
  "packageRules": [
    {
      "managers": ["npm"],
      "commitMessage": "{{{commitMessagePrefix}}} {{{commitMessageTopic}}}{{{commitMessageExtra}}} {{{commitMessageSuffix}}}",
      "commitMessageTopic": "{{depName}}",
      "commitMessageExtra": "{{#if isMajor}} v{{{newMajor}}}{{else}}{{#if isSingleVersion}}@{{{toVersion}}}{{else}}@{{{newValue}}}{{/if}}{{/if}}"
    },
    {
      "packagePatterns": [
        "*"
      ],
      "rangeStrategy": "replace",
      "semanticCommitType": "housekeeping"
    },
    {
      "depTypeList": [
        "devDependencies"
      ],
      "rangeStrategy": "update-lockfile",
      "semanticCommitScope": "dev-deps",
      "automerge": true,
      "major": {
        "automerge": false
      }
    },
    {
      "depTypeList": [
        "peerDependencies"
      ],
      "rangeStrategy": "widen",
      "semanticCommitScope": "peer-deps"
    },
    {
      "packagePatterns": ["^react"],
      "enabled": false
    }
  ],
  "ignorePaths": ["swagger-ui-dist-package/package.json"],
  "ignoreDeps": ["release-it", "@release-it/conventional-changelog"],
  "prConcurrentLimit": 3
}
```

</details>

Maybe they could use some combination of the `commitMessage...` options? I don't know enough about this to help out but maybe @rarkins or @viceice can cobble something together?

See [Renovate docs, commitmessage](https://docs.renovatebot.com/configuration-options/#commitmessage) section.

**Comment by rarkins** [comment]
The reason we don't want people editing commitMessage directly is because:
- commitMessage is used to create prTitle
- prTitle is very important for detecting duplications of PRs

So two possible outcomes:
- Are there requirements people have for commitMessage which can't be achieved by using its subcomponents? <- undeprecate commitMessage
- Is there a way to allow any commit message people want but we set prTitle to what *we* want? I actually think more people care about the PR title than the commit message

**Comment by rarkins** [comment]
I agree to not deprecate commit message editing - it's PR title which is the most important. So while we may default them to the same, we need to retain editing of commitMessage even if not prTitle
